src/diffusers/configuration_utils.py

```diff
@@ -156,6 +162,18 @@ def save_config(self, save_directory: Union[str, os.PathLike], push_to_hub: bool
         self.to_json_file(output_config_file)
         logger.info(f"Configuration saved in {output_config_file}")
 
+        if push_to_hub:
+            commit_message = kwargs.pop("commit_message", None)
+            repo_id = kwargs.pop("repo_id", save_directory.split(os.path.sep)[-1])
+            repo_id = create_repo(repo_id, exist_ok=True, **kwargs).repo_id
+
+            self._upload_folder(
```

Comment:
`token` should be passed explicitly here IMO (same for `create_repo`). Otherwise it'll work only for logged in users